### PR TITLE
WalletRegistry governance params

### PR DESCRIPTION
**Pull request by dimpar**
Refs #2758

Depends on [#2792](https://github.com/keep-network/keep-core/pull/2792)

Adding WalletRegistryGovernance contract for updating governance params:
- walletOwner
- maliciousDkgResultSlashingAmount
- maliciousDkgResultNotificationRewardMultiplier
- resultChallengePeriodLength
- dkgResultSubmissionTimeout
- submitterPrecedencePeriodLength

### CODE REVIEW

**Review by nkuba — CHANGES_REQUESTED** [review]

File: solidity/ecdsa/test/fixtures/index.ts [hunk 1]
```diff
@@ -30,7 +30,7 @@ export const dkgState = {
   CHALLENGE: 4,
 }
 
-export const params = {
+export const dkgParams = {
```
[the remark this answers is not in the document]
Comment (dimpar) [reply solidity/ecdsa/test/fixtures/index.ts]: f396c5b

File: solidity/ecdsa/contracts/WalletRegistry.sol [hunk 2]
```diff
@@ -324,4 +376,12 @@ contract WalletRegistry is Ownable {
     {
         return wallets.registry[walletID];
     }
+
+    function dkgResultParameters()
```
[the remark this answers is not in the document]
Comment (dimpar) [reply solidity/ecdsa/contracts/WalletRegistry.sol]: a5f236e

File: solidity/ecdsa/contracts/WalletRegistryGovernance.sol [hunk 3]
```diff
@@ -0,0 +1,363 @@
...
+        uint256 dkgResultSubmissionEligibilityDelay,
+        uint256 timestamp
+    );
+    event DkgResultSubmissionEligibilityDelayUpdated(
+        uint256 dkgResultSubmissionEligibilityDelay
+    );
+
+    /// @notice Reverts if called before the governance delay elapses.
+    /// @param changeInitiatedTimestamp Timestamp indicating the beginning
+    ///        of the change.
```
[the remark this answers is not in the document]
Comment (dimpar) [reply solidity/ecdsa/contracts/WalletRegistryGovernance.sol:84]: 51c0273

File: solidity/ecdsa/contracts/WalletRegistryGovernance.sol [hunk 4]
```diff
@@ -0,0 +1,363 @@
...
+        )
+    {
+        emit DkgResultSubmissionEligibilityDelayUpdated(
+            newDkgResultSubmissionEligibilityDelay
+        );
+        // slither-disable-next-line reentrancy-no-eth
+        walletRegistry.updateDkgParameters(
+            walletRegistry
+                .dkgResultParameters()
+                .resultSubmissionEligibilityDelay,
```
[the remark this answers is not in the document]
Comment (dimpar) [reply solidity/ecdsa/contracts/WalletRegistryGovernance.sol]: 3d6225f

File: solidity/ecdsa/contracts/WalletRegistry.sol [hunk 5]
```diff
@@ -144,17 +155,58 @@ contract WalletRegistry is Ownable {
...
+    ///      wallet registry governance contract. The caller is responsible for
+    ///      validating parameters.
+    /// @param _resultChallengePeriodLength New DKG result challenge period
+    ///        length
+    /// @param _resultSubmissionEligibilityDelay New DKG result submission
+    ///        eligibility delay
+    function updateDkgParameters(
+        uint256 _resultChallengePeriodLength,
+        uint256 _resultSubmissionEligibilityDelay
     ) external {
```
[the remark this answers is not in the document]
Comment (dimpar) [reply solidity/ecdsa/contracts/WalletRegistry.sol]: 9754e37

File: solidity/ecdsa/test/WalletRegistryGovernance.test.ts [hunk 6]
```diff
@@ -0,0 +1,664 @@
...
+  const initialMaliciousDkgResultSlashingAmount = to1e18(50000)
+  const initialMaliciousDkgResultNotificationRewardMultiplier = 100
+
+  before("load test fixture", async () => {
+    // eslint-disable-next-line @typescript-eslint/no-extra-semi
+    ;({ walletRegistry, governance, thirdParty } = await waffle.loadFixture(
+      walletRegistryFixture
+    ))
+
+    const WalletRegistryGovernance = await ethers.getContractFactory(
```
[the remark this answers is not in the document]
Comment (dimpar) [reply solidity/ecdsa/test/WalletRegistryGovernance.test.ts]: 6837f64

File: solidity/ecdsa/test/WalletRegistryGovernance.test.ts [hunk 7]
```diff
@@ -0,0 +1,966 @@
...
+      })
+    })
+
+    context("when the update value is at least 10", () => {
+      it("should accept the value", async () => {
+        await createSnapshot()
+
+        await walletRegistryGovernance
+          .connect(governance)
+          .beginDkgResultChallengePeriodLengthUpdate(10)
```
Comment (nkuba) [inline solidity/ecdsa/test/WalletRegistryGovernance.test.ts:491]: We could use `to.not.be.reverted` for verification (in other places too).

```suggestion
        await expect(
          walletRegistryGovernance
            .connect(governance)
            .beginDkgResultChallengePeriodLengthUpdate(10)
          ).to.not.be.reverted
```

**Review by nkuba — APPROVED** [review]
I don't see any blockers. We can handle my comments in a follow-up PR.

**Comment by dimpar** [comment]
> I don't see any blockers. We can handle my comments in a follow-up PR.

Follow up PR is ready https://github.com/keep-network/keep-core/pull/2820

**Pull request merged**